### PR TITLE
Added object and endpoints for managed credentials

### PR DESCRIPTION
**Pull request by yoshizzle**
GET all credentials and specific credentials by ID
POST to create a new credential
PUT to update an existing credential
POST to password to change an existing credential's password
POST to revoke to remove a credential

### CODE REVIEW

File: docs/src/data/endpoints/managed.yaml [hunk 1]
```diff
@@ -175,3 +175,104 @@ endpoints:
...
+          Registers a new credential.
+        params:
+          label:
+            type: String
+            limit: 2-75 characters
+            description: >
+              A unique label describing the credential being added.
+          username:
+            type: String
+            description: >
```
Comment (SzySteve) [inline docs/src/data/endpoints/managed.yaml]: optional: True

File: docs/src/data/endpoints/managed.yaml [hunk 2]
```diff
@@ -175,3 +175,104 @@ endpoints:
...
+              An optional login username.
+          password:
+            type: String
+            description: >
```
Comment (SzySteve) [inline docs/src/data/endpoints/managed.yaml:207]: optional: True

File: docs/src/data/endpoints/managed.yaml [hunk 3]
```diff
@@ -175,3 +175,104 @@ endpoints:
...
+      GET:
+        response: ManagedCredential
+        description: >
+          Returns information about this credential.
+        examples:
+          curl: |
+            curl -H "Authorization: Bearer $TOKEN" \
+                https://$api_root/$version/managed/credential/$credential_id
+      PUT:
+        description: >
```
Comment (SzySteve) [inline docs/src/data/endpoints/managed.yaml:235]: Should note what params are optional?

**Review by SzySteve — CHANGES_REQUESTED** [review]

File: docs/src/data/objects/managedcredential.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      A unique Label describing the credential.
+  username:
+    type: String
+    value: myusername
+    description: >
+      An optional login username.
+  last_decryption:
+    type: Object
+    description: >
+      The username and datetime of the last successful login using this credential.
```
Comment (Dorthu) [inline docs/src/data/objects/managedcredential.yaml]: not sure if these are "successful" decryptions only

File: docs/src/data/endpoints/managed.yaml (hunk 3 above)
Comment (Dorthu) [reply docs/src/data/endpoints/managed.yaml:235]: I don't think so, PUTs are at least partially generated from the object I think

File: docs/src/data/endpoints/managed.yaml (hunk 2 above)
Comment (yoshizzle) [reply docs/src/data/endpoints/managed.yaml:207]: password is required.

File: docs/src/data/objects/managedcredential.yaml (hunk 4 above)
Comment (yoshizzle) [reply docs/src/data/objects/managedcredential.yaml]: I will remove the word successful.

File: docs/src/data/endpoints/managed.yaml (hunk 3 above)
Comment (yoshizzle) [reply docs/src/data/endpoints/managed.yaml:235]: Not sure what you mean. I don't think we do that typically.

**Review by Dorthu — APPROVED** [review]

**Review by SzySteve — APPROVED** [review]

**Pull request merged**